src/mds/mdstypes.h

```diff
@@ -73,6 +73,13 @@
 #define MDS_TRAVERSE_DISCOVER      2    // skips permissions checks etc.
 #define MDS_TRAVERSE_DISCOVERXLOCK 3    // succeeds on (foreign?) null, xlocked dentries.
 
+#define WORM_ENABLE  1     // worm feature enbale
+#define WORM_IS_ROOT 2     // worm root dir
+#define WORM_RETAIN  4     // the worm file is in retained state
+#define WORM_EXPIRE  8     // the worm file is in expired state
+#define WORM_RETEN_PERIOD_UNLIMIT 16  //the worm file retention period is unlimit
+#define MAX_MIN_RETEN_PERIOD 946080000 // 30 years
+#define MIN_MAX_RETEN_PERIOD 2207520000 // 70 years
```

Comment:
Our customers require a minimum retention period that can be set from 0 to 30 years, and a maximum retention period from 0 to 70 years. And to avoid other people being confused like you, I will remove them and put the restrictions on the configuration of our own management system.